includes/abstracts/abstract-wc-log-handler.php

```diff
@@ -0,0 +1,141 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+if ( ! class_exists( 'WC_Logger' ) ) {
+	include_once( dirname( dirname( __FILE__ ) ) . '/class-wc-logger.php' );
+}
+
+/**
+ * Abstract WC Log Handler Class
+ *
+ * @version        1.0.0
+ * @package        WooCommerce/Abstracts
+ * @category       Abstract Class
+ * @author         WooThemes
+ */
+abstract class WC_Log_Handler {
+
+	/**
+	 * Minimum log level this handler will process.
+	 *
+	 * @var int 0-8 minimum level severity for handling log entry.
+	 * @access private
+	 */
+	protected $threshold;
+
+	/**
+	 * Log levels by severity.
+	 *
+	 * @var array
+	 * @access private
+	 */
+	protected static $log_levels = array(
+		WC_Logger::DEBUG     => 0,
+		WC_Logger::INFO      => 1,
+		WC_Logger::NOTICE    => 2,
+		WC_Logger::WARNING   => 3,
+		WC_Logger::ERROR     => 4,
+		WC_Logger::CRITICAL  => 5,
+		WC_Logger::ALERT     => 6,
+		WC_Logger::EMERGENCY => 7,
+	);
+
+	/**
+	 * Constructor for log handler.
+	 *
+	 * @param array $args {
+	 *     @type string $threshold Optional. Default 'emergency'. Sets the log severity threshold.
+	 *         emergency|alert|critical|error|warning|notice|info|debug
+	 * }
+	 */
+	public function __construct( $args = array() ) {
+
+		$args = wp_parse_args( $args, array(
+			'threshold' => WC_Logger::DEBUG,
+		) );
+
+		$this->set_threshold( $args['threshold'] );
+	}
+
+	/**
+	 * Handle a log entry.
+	 *
+	 * @param int $timestamp Log timestamp.
+	 * @param string $level emergency|alert|critical|error|warning|notice|info|debug
+	 * @param string $message Log message.
+	 * @param array $context Additional information for log handlers.
+	 *
+	 * @return bool True if log entry should bubble to further loggers.
+	 */
+	abstract public function handle( $timestamp, $level, $message, $context );
+
+	/**
+	 * Set handler severity threshold.
+	 *
+	 * @param string $level emergency|alert|critical|error|warning|notice|info|debug
+	 */
+	public function set_threshold( $level ) {
+		$level = apply_filters( 'woocommerce_log_handler_set_threshold', $level, get_class( $this ) );
```

Comment:
Name or ID would likely be more useful because they can refer to a specific instance, whereas the class name is shared. At this point `name` makes sense because there is no requirement for it to be unique. The name would need to be provided at instantiation, the `WC_Log_Handler` abstract class can be modified to accept a `name`, and use `get_class( $this )` as a default.